src/solvers/refinement/refine_arrays.cpp

```diff
@@ -55,7 +56,9 @@ void bv_refinementt::arrays_overapproximated()
     if(current.id()==ID_implies)
     {
       implies_exprt imp=to_implies_expr(current);
-      assert(imp.operands().size()==2);
+      DATA_INVARIANT(
+        imp.operands().size()==2,
+        string_refinement_invariantt("implies has two operands"));
```

Comment:
A clearer message would be `implies must have two operands`. (and it would consistent with the other messages)